`virtio-rng` documentation (#3589)
firecracker-microvm/firecracker: docs/snapshotting/random-for-clones.md

## Background

The Linux kernel exposes three main `RNG` interfaces to userspace: the
`/dev/random` and `/dev/urandom` special devices, and the `getrandom` syscall,
which are described in the [random(7) man page][1]. Moreover, Firecracker
supports the [`virtio-rng`](../entropy.md) device which can provide additional
entropy to guest VMs. The source of the random bytes provided by `virtio-rng`
is `getrandom` or `/dev/urandom` after ensuring that the entropy pool has been

> initialized.

Author: Here we are speaking about the host. `/dev/hwrng` is a guest device in this context.

> which are described in the [random(7) man page][1]. Moreover, Firecracker
> supports the [`virtio-rng`](../entropy.md) device which can provide additional
> entropy to guest VMs. The source of the random bytes provided by `virtio-rng`
> is `getrandom` or `/dev/urandom` after ensuring that the entropy pool has been
> initialized.

Reviewer: Ahhh, my bad. I thought the last sentence refers to where the guest can gets its virtio-rng randomness from. Could we clarify the "host" part of this? Maybe something like 

```suggestion
which are described in the [random(7) man page][1]. Moreover, Firecracker
supports the [`virtio-rng`](../entropy.md) device which can provide additional
entropy to guest VMs. It draws its random bytes from the host kernel via `getrandom` 
or `/dev/urandom` after ensuring that the entropy pool has been initialized.
```

instructions which are present on Ivy Bridge or newer Intel processors
and enabled in a Firecracker guest) are mixed with each result. Extra
bits are mixed in both when reading random values, and in conjunction
with entropy related events such as interrupts. Moreover, the guest kernel

> will eventually receive fresh entropy from `virtio-rng`, if attached. There are

Reviewer 2: nit: this sentence is rather long and hard to fit into cache. Is it possible to organise it somehow, maybe via bullet points for the questions?

Author: I agree. I re-worked this. Please take a look.

> will eventually receive fresh entropy from `virtio-rng`. There are two

Author: you're right!
